### DISCUSSION TITLE
DagsterInvalidSubsetError when asset exists

### DISCUSSION THREAD

**Discussion by EtienneT** [dagster-io/dagster#21104] [Q&A] [answered]
I am getting the following error message only on one partition for a new asset I created.  This happens even when I try to manually trigger materialization for the asset from the UI.  So obviously the asset is in the definition since I can access it in the UI.

dagster._core.errors.DagsterInvalidSubsetError: AssetKey(s) ['price_daily'] were selected, but no AssetsDefinition objects supply these keys. Make sure all keys are spelled correctly, and all AssetsDefinitions are correctly added to the `Definitions`.

Some partition had this error, I triggered a manual materialization and the run went through successfully, but for one partition, no matter how many try I manually trigger it, it always fails with this error.

**Comment by garethbrickman** [comment, marked as answer]
Resolved in https://github.com/dagster-io/dagster/issues/21152
